checkov/circleci_pipelines/checks/SuspectCurlInScript.py

```diff
@@ -0,0 +1,33 @@
+from __future__ import annotations
+from typing import Any
+
+from checkov.circleci_pipelines.base_circleci_pipelines_check import BaseCircleCIPipelinesCheck
+from checkov.common.models.enums import CheckResult
+from checkov.yaml_doc.enums import BlockType
+
+class SuspectCurlInScript(BaseCircleCIPipelinesCheck):
+    def __init__(self) -> None:
+        name = "Suspicious use of curl in run task"
+        id = "CKV_CIRCLECIPIPELINES_7"
+        super().__init__(
+            name=name,
+            id=id,
+            block_type=BlockType.ARRAY,
+            supported_entities=['jobs.*.steps[]']
+        )
+
+    def scan_conf(self, conf: dict[str, Any]) -> tuple[CheckResult, dict[str, Any]]:
+        if "run" not in conf:
+            return CheckResult.PASSED, conf
+        run = conf.get("run", "")
+        if type(run) == dict:
+            run = run.get("command", "")
+        if "curl" in run:
+            badstuff = ['curl', 'POST']
+            lines = run.split("\n")
```

Comment:
```suggestion
        if isinstance(run, dict):
            command = run.get("command", "")
        if "curl" in run:
            badstuff = ('curl', 'POST')
            lines = command.split("\n")
```
same here + change the list to a tuple, because the content is static and tuple has better performance for this scenario.